litex/workshop_rot13.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# This variable defines all the external programs that this module
+# relies on.  lxbuildenv reads this variable in order to ensure
+# the build will finish without exiting due to missing third-party
+# programs.
+LX_DEPENDENCIES = ["icestorm", "yosys", "nextpnr-ice40"]
+#LX_CONFIG = "skip-git" # This can be useful for workshops
+
+# Import lxbuildenv to integrate the deps/ directory
+import os,os.path,shutil,sys,subprocess
+sys.path.insert(0, os.path.dirname(__file__))
+import lxbuildenv
+
+# Disable pylint's E1101, which breaks completely on migen
+#pylint:disable=E1101
+
+from migen import *
+from migen.genlib.resetsync import AsyncResetSynchronizer
+
+from litex.soc.integration.soc_core import SoCCore
+from litex.soc.integration.builder import Builder
+from litex.soc.interconnect.csr import AutoCSR, CSRStatus, CSRStorage, CSRField
+
+from litex_boards.partner.targets.fomu import BaseSoC, add_dfu_suffix
+
+from valentyusb.usbcore import io as usbio
+from valentyusb.usbcore.cpu import dummyusb
+
+import argparse
+
+# ROT13 input CSR. Doesn't need to do anything special.
+class FomuROT13In(Module, AutoCSR):
+    def __init__(self):
+        self.csr = CSRStorage(8)
+
+# ROT13 output CSR, plus the wiring to automatically create the output from the input CSR.
+class FomuROT13Out(Module, AutoCSR):
+    def __init__(self, rot13_in):
+        self.csr = CSRStorage(8)
+        self.sync += If( # A-M, a-m
```

Comment:
I think this could do with some more explanation :-)

Reply:
Added comments in 5a3c933, let me know if that's what you're looking for.

Reply:
Yeap starting to look good! I do think you might want a comment which says "we only rotate the alphabetic characters" (what about the numbers? :-) and maybe a link to the ROT13 wikipedia page?

Could be good to add a test bench in the future too....

From https://en.wikipedia.org/wiki/ROT13 it looks like [`memfrob()` in the GNU C library](https://www.gnu.org/software/libc/manual/html_node/Obfuscating-Data.html) could be an excellent target for this type of acceleration :-P
> The GNU C library, a set of standard routines available for use in computer programming, contains a function—memfrob()[15]—which has a similar purpose to ROT13, although it is intended for use with arbitrary binary data. The function operates by combining each byte with the binary pattern 00101010 (42) using the exclusive or (XOR) operation.

Reply:
Linked the wiki page, added a brief explanation of the overall cipher, mentioned that it's limited to the English alphabet.